Divine/Enums/LogLevel.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Divine.Enums
+{
+    public enum LogLevel
+    {
+        SILENT = 0,

```

Comment:
I'm not sure I understand the ordering here. In traditional log level ordering, more important/serious messages go to the "higher" levels and less important/debug messages go to the "lower" levels. In our case, FATAL is 2 levels after WARN, however displaying fatal errors is clearly a higher priority than warnings. Therefore I'd say that the correct order should be:
INFO -> FATAL -> ERROR -> WARN -> DEBUG

Reply:
I disagree with your order, so I'll just copy the order from log4j.

Level | Value
--- | ---
OFF | 0
FATAL | 100
ERROR | 200
WARN | 300
INFO | 400
DEBUG | 500
TRACE | 600
ALL | MAX_VALUE

Reply:
Not that I have much stake in this discussion, but all logging I have ever used in my career is pretty much the same as what Fire listed and is also what I would find most useful.

Reply:
Yeah, the LOG4J order seems fine.

Reply:
- LogFatal now requires an errorCode parameter and does not output the stack.
- LogTrace added and outputs the stack separately.
- All instances of LogFatal now return an exit code: `(int)LogLevel.FATAL + errorCode`
- All instances of LogFatal updated with an errorCode of 1 for user errors and 2 for program errors.

Not committed yet.